### ISSUE TITLE
Registering callbacks/hooks/picker providers with a priority via annotations is broken

### ISSUE THREAD

**Issue by m-vo** [contao/contao#2570]
**Affected version(s)**

Contao 4.10

**Description**
This [commit](https://github.com/contao/contao/commit/e68ed7fbf66d2c6d1cfa3a880f96fc3501592b2b) from #2476 breaks defining callbacks, hooks and picker providers with a priority. This will fail silently and the elements won't be registered at all. :scream: 

Here is an example:
```
[Type Error] Attribute "priority" of @Callback declared on method Contao\CoreBundle\EventListener\DataContainer\ContentCompositionListener::generateArticleForPage() expects a(n) int|null, but got integer.
```

This is because of this check in the doc parser that now fails because `'int' !== 'int|null'`: https://github.com/doctrine/annotations/blob/984d7dc8d61c3566768379fe25aa531b54b8130d/lib/Doctrine/Common/Annotations/DocParser.php#L902

Doctrine Annotations should probably handle this but we should imo fix this asap as this renders lots of things unusable in the current 4.10.

**Comment by m-vo** [comment]
For a quick fix see #2572.

**Comment by aschempp** [comment]
Closing in favor of #2572
